pennylane/templates/subroutines/qpe.py

```diff
@@ -141,4 +141,14 @@ def expand(self):
 
             qml.templates.QFT(wires=self.estimation_wires).inv()
 
+        if self.inverse:
+            tape.inv()
+
         return tape
+
+    def adjoint(self):
+        adjoint_op = QuantumPhaseEstimation(
+            *self.parameters, target_wires=self.target_wires, estimation_wires=self.estimation_wires
+        )
+        adjoint_op.inverse = not self.inverse
```

Comment:
Out of curiosity, why are we doing ``adjoint_ob.inverse = not self.inverse`` here?

Reply:
This was a major problem I found. If we don't set that ( and always set `True` for example), the inverse of the inverse would not update properly. The correct thing to do is to change True to False and False to True 😄